programs/Makefile

```diff
@@ -84,6 +84,10 @@ default: zstd
 
 all: zstd zstd32 fullbench fullbench32 fuzzer fuzzer32 zbufftest zbufftest32 paramgrill datagen
 
+zstd_clang: $(ZSTD_FILES) $(ZSTD_FILES_LEGACY) $(ZSTDDIR)/zbuff.c $(ZSTDDIR)/zdict.c $(ZSTDDIR)/divsufsort.c \
+        zstdcli.c fileio.c bench.c xxhash.c datagen.c dibio.c
+	clang.exe --target=x86_64-w64-mingw32 -Werror -Wconversion -Wno-sign-conversion $(FLAGS) -DZSTD_LEGACY_SUPPORT=$(ZSTD_LEGACY_SUPPORT) $^ -o $@$(EXT)
```

Comment:
I don't think you need to put that in the `Makefile` then.
You could have your own local batch file, and launch the compilation process like this : 
`make zstd CC="clang.exe --target=x86_64-w64-mingw32" `

You can also use MOREFLAGS the same way if you want to add specific flags not present by default in the `Makefile`